File: app/src/main/java/com/icesoft/msdb/android/model/Event.java
package com.icesoft.msdb.android.model;

import java.util.Optional;

import lombok.Data;

@Data
public class Event extends BaseModel {
    private Long id;
    private String name;
    private Boolean rally;
    private Boolean raid;

    public boolean isRally() {
        return Optional.ofNullable(rally).orElse(false);
    }

    public boolean isRaid() {
        return Optional.ofNullable(raid).orElse(false);
    }
}

File: app/src/main/java/com/icesoft/msdb/android/model/Event.kt
package com.icesoft.msdb.android.model

import java.util.*

data class Event(
    val id: Long? = 0,
    val name: String? = null,
    val rally: Boolean? = false,
    val raid: Boolean? = false
): BaseModel() {
    fun isRally(): Boolean {
        return Optional.ofNullable(rally).orElse(false)
    }

    fun isRaid(): Boolean {
        return Optional.ofNullable(raid).orElse(false)
    }
}
